tokio/src/sync/mpsc/bounded.rs

```diff
@@ -1278,6 +1410,47 @@ impl<T> fmt::Debug for Permit<'_, T> {
     }
 }
 
+// ===== impl PermitIterator =====
+
+impl<'a, T> Iterator for PermitIterator<'a, T> {
+    type Item = Permit<'a, T>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.n == 0 {
+            return None;
+        }
+
+        self.n -= 1;
+        Some(Permit { chan: self.chan })
+    }
+}
```

Comment:
Please add an implementation of `size_hint` as well. I think it also makes sense to implement `ExactSizeIterator`, since we know the exact length.